core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -215,7 +247,12 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
 
   def addListeners(listenersAdded: Seq[EndPoint]): Unit = synchronized {
     info(s"Adding listeners for endpoints $listenersAdded")
-    createAcceptorAndProcessors(config.numNetworkThreads, listenersAdded)
+    val (controlPlaneListenersAdded, dataPlaneListenersAdded) = listenersAdded.partition { endpoint =>
+        config.controlPlaneListenerName.isDefined && endpoint.listenerName.value() == config.controlPlaneListenerName.get }
+    if (dataPlaneListenersAdded.nonEmpty)
+      createAcceptorAndProcessors(config.numNetworkThreads, dataPlaneListenersAdded, dataRequestChannel, dataProcessors, false)
+    if (controlPlaneListenersAdded.nonEmpty)
+      createAcceptorAndProcessors(1, controlPlaneListenersAdded, controlPlaneRequestChannel, controlPlaneProcessors, true)
```

Comment:
should we have a check that a control plane listener is already defined, since there should be only one controlPlaneListener, no?